apps/tailwind-components/components/button/FilterWellContainer.vue

```diff
@@ -0,0 +1,160 @@
+<script lang="ts" setup>
+import { ref, useTemplateRef, watch, onMounted, onBeforeUnmount } from "vue";
+
+const buttonContainer = useTemplateRef<HTMLDivElement>("filterButtonContainer");
+const widthObserver = ref();
+const showButtons = ref<boolean>(false);
+const hasFilterWellButtons = ref<boolean>(false);
+const filterButtons = ref<HTMLCollection>();
+const focusCounter = ref<number>(0);
+const minFocusCounter = ref<number>(0);
+const maxFocusCounter = ref<number>(0);
+
+const scrollConfig = {
+  behavior: "smooth",
+  block: "nearest",
+  inline: "center",
+} as ScrollOptions;
+
+const observerConfig = {
+  childList: true,
+  subtree: true,
+  attributes: false,
+  characterData: false,
+};
+
+const emit = defineEmits(["clear"]);
+
+function observeScrollWidth() {
+  if (
+    buttonContainer.value &&
+    buttonContainer.value.scrollWidth > buttonContainer.value.offsetWidth
+  ) {
+    showButtons.value = true;
+  } else {
+    showButtons.value = false;
+  }
+
+  updateFilterButtons();
+}
+
+function updateFilterButtons() {
+  filterButtons.value = buttonContainer.value?.children as HTMLCollection;
+  maxFocusCounter.value = filterButtons.value.length;
+
+  if (buttonContainer.value) {
+    hasFilterWellButtons.value = buttonContainer.value.children.length > 1;
+  }
+}
+
+function addFilterButtonClass(button: HTMLButtonElement) {
+  button.classList.add("bg-button-filter-hover");
+  button.classList.add("border-button-filter-hover");
+}
+
+function removeFilterButtonClass(button: HTMLButtonElement) {
+  button.classList.remove("bg-button-filter-hover");
+  button.classList.remove("border-button-filter-hover");
+}
```

Comment:
I've added these functions to visually focus buttons, but I've temporarily disabled it. I do not want to use `button.focus()` as this would be disruptive for other types of navigation.

Reply:
This isn't needed any more